enhancements/microshift/microshift-updateability-ostree.md

```diff
@@ -356,12 +354,13 @@ N/A
 
 ### General overview of actions order
 
-1. System boots
-1. "Pre run" phase (failure will block "run" phase)
-   - Data is backed up, restore, or (in special cases) cleared.
-   - If needed, data is migrated to newer version.
-1. "Run" phase
-   - All MicroShift components start
+1. MicroShift binary starts
+1. "Data management" or "backup and/or restore" phase (failure will block next phases)
+   - Data is backed up, restored, and/or (in special cases) cleared to allow for fresh start.
+1. Version metadata in data directory and version of MicroShift's executable are compared.
+   - Invalid upgrade paths cause MicroShift to exit with error
+   - If the version change is valid, version in data directory is updated with executable's version.
+1. All MicroShift components start - this includes Kubernetes Storage Migrator
```

Comment:
Maybe an addition of a state digram might help visualize this flow as well?

```mermaid
flowchart LR
    A[MicroShift Starts] --> B(Data Management)
    B --> C{Backup/Restore}
    C -->|Failure| D[MicroShift Stop]
    C -->|Success| F{Is Version \nUpgradeable?}
    F -->|Failure| D
    F -->|Success| G[All Components Start]  
```

Reply:
+1 for diagrams

Reply:
I included a flowchart, but took a slightly different approach. PTAL

Reply:
Looks good, I like it 👍